massgov/mds-licensing
# Implement registration page and JWT endpoint as local code

This PR pulls the registration and JWT endpoints for licensing into this repository, where they had previously been implemented as separate Cloudfront origins based on routing logic on the distribution.   I believe this is important because it allows us to visualize the extent of the application just by looking at the repository (depending on another, separately deployed APIG and S3 bucket is not ideal because it creates invisible dependencies).

So the following actions were taken:
* Copy the JWT generation endpoint into a Lambda in this repository instead of proxying to the cognitoAuthentication APIG.
* Copy the registration page out of S3 and put it into a very simple Lambda that serves the page.  Along the way, I streamlined the dependencies:
    * Switch to the new version of `amazon-cognito-identity-js`, which has slightly different semantics (the previous version has been deprecated).
    * Use CDN versions of all external dependencies.
    * Inline the app javascript in the registration page to avoid needing to serve up javascript files too.
* Use environment variables for the Cognito pool and client IDs anywhere they're used.  

Please note that `/registration.html` and `/return-jwt` operate the same, but have been moved under the stage (`/v1`) as a result of this change.  To ensure backward compatibility, we will still need to do some routing in the Cloudfront domains:

* `/return-jwt/*` -> `/v1/return-jwt/*`
* `/registration.html` -> `/v1/registration.html`

I've added a note about this to DP-11208 and DP-11209 so we don't forget to do it. 

### Steps to test

1. Visit https://dwrtlf3ife.execute-api.us-east-1.amazonaws.com/v1/register.html and verify that you are able to create a new account.
2. As an admin, confirm the new account.
3. Use Postman to hit `https://dwrtlf3ife.execute-api.us-east-1.amazonaws.com/v1/return-jwt/license-validation-api/access-token` with the correct `username_val` and `password_val`, and verify you're able to obtain a JWT.  
4. Verify that you're able to use the JWT from step 3 for the boards endpoint: `https://dwrtlf3ife.execute-api.us-east-1.amazonaws.com/v1/licenses/boards/agencies/EOPSS`
